Switch to electron builder.

bayleedev opened the pull request:
This enables us to start signing our builds and creating builds for more
platforms.

However, all this does is change us TO electron builder with no behavior
changes.

I also updated some documentation, the changelog and the current
version. Once this is merged I'll do a deployment of v0.4.0.

twang2218 commented on package.json:
```diff
@@ -1,10 +1,54 @@
+  },
+  "directories": {
+    "buildResources": "resources"
+  },
```
`directories` should be moved to `build` section, otherwise the following warning will be shown during the build:

```
⚠️  "directories" in the root is deprecated, please specify in the "build"
```

bayleedev replied:
Updated! [:

twang2218 commented on package.json:
```diff
@@ -1,10 +1,54 @@
     "cucumber": "^1.2.2",
+    "electron-builder": "^11.5.2",
     "electron-mocha": "^1.0.0",
     "electron-prebuilt": "~1.3.1",
```
`electron-prebuilt` should be updated to `electron`, as `electron-prebuilt` is deprecated now.

bayleedev replied:
Should this be saved for the next time we upgrade electron?

twang2218 replied:
`electron-prebuilt` stopped updating since the end of 2016, that's the reason I thought it would be good to update to the latest the package name. However, I'm not quite sure how much breaking changes between electron 1.3 and 1.4, so maybe it's a better idea to leave it to another PR.

bayleedev replied:
I'll try switching without upgrading and see if that causes any issues. My guess is that either the creating a release or the cucumber specs will fail.

bayleedev replied:
Seems to work, I'll update [:

twang2218 commented on package.json:
```diff
@@ -1,12 +1,56 @@
+    "publish": [
+      "github"
+    ],
+    "appId": "TinyTacoTeam.ZazuApp",
```
`productName` is missing here. Previously, the `productName` is `Zazu`, and without it, the name become `zazu`, which are all lower case. This affects some other fields in the generated app, such as `CFBundleDisplayName`:

Previously, the `Info.plist` contains:

```xml
        <key>CFBundleDisplayName</key>
        <string>Zazu</string>
        <key>CFBundleExecutable</key>
        <string>Zazu</string>
```

And now, it becomes:

```xml
    <key>CFBundleDisplayName</key>
    <string>zazu</string>
    <key>CFBundleExecutable</key>
    <string>zazu</string>
```

bayleedev replied:
Wow, that's an awesome catch.

twang2218 commented on package.json:
```diff
@@ -1,12 +1,57 @@
+  "version": "0.4.0",
+  "build": {
+    "productName": "Zazu",
+    "appId": "TinyTacoTeam.ZazuApp",
```
`appId` should be `com.tinytacoteam.zazu`, as it will become the identifier of the App,

on `master` branch, the generated `Info.plist` contains:

```xml
        <key>CFBundleIdentifier</key>
        <string>com.tinytacoteam.zazu</string>
```

And in this PR, it become:

```xml
    <key>CFBundleIdentifier</key>
    <string>TinyTacoTeam.ZazuApp</string>
```

Commit:
Switch to electron builder.

This enables us to start signing our builds and creating builds for more
platforms.

However, all this does is change us TO electron builder with no behavior
changes.

I also updated some documentation, the changelog and the current
version. Once this is merged I'll do a deployment of v0.4.0.

bayleedev commented on changelog.md:
```diff
@@ -1,3 +1,23 @@
+## 0.3.3
```
Should be 0.4.0
